[[ MinStackFileVersion ]] Implement minStackFileVersion stack prop

montegoulding opened the pull request:
This patch implements a read only property to determine what
`stackFileVersion` may be used to safely save a stack without
data loss.

Required to fix [this](http://quality.livecode.com/show_bug.cgi?id=18175) with IDE patch forthcoming.

livecodeali commented on docs/dictionary/property/minStackFileVersion.lcdoc:
```diff
@@ -0,0 +1,36 @@
+Platforms: desktop, server, mobile
+
+Example:
+set the stackFileVersion to the minStackFileVersion of this stack
```
`stackFileVersion` is deprecated I think, so it would be better to use

```
save this stack with format (the minStackFileVersion of this stack)
```

as the example

livecodeali commented on docs/dictionary/property/minStackFileVersion.lcdoc:
```diff
@@ -0,0 +1,36 @@
+may not support <save|saving> some newer features. For example, stacks
+with widgets must be saved with at least a stack file version of 8.0.
+
+References: property (glossary), stackFileVersion (property), 
```
I think all the references to `stackFileVersion` should be changed to reference the `stack version` glossary entry.

montegoulding committed:
[[ MinStackFileVersion ]] Implement minStackFileVersion stack prop

This patch implements a read only property to determine what
stackFileVersion may be used to safely save a stack without
data loss.

livecodeali commented:
@livecode-vulcan review ok 1951f32

livecode-vulcan commented:
💙 review by @livecodeali ok 1951f32a4b2d4c700e8f5912a7940fe5360332e7

livecode-vulcan commented:
😎 test success 1951f32a4b2d4c700e8f5912a7940fe5360332e7
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/88)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/88)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/88)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/88)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/88)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/88)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/88)
